File: Data Structures & Algorithms/divide-array-into-equal-pairs/submission-2.py
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        freq= {}
        for i in nums :
            freq[i] = freq.get(i,0) + 1
        
        for key,value in freq.items() :
            if value %2 != 0 :
                return False
        
        return True

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        
    unordered_map<int, int> freq;
    for (int num : nums) {
        freq[num]++;
    }

    map<int, set<int>, greater<int>> freqToNums;

    for (auto &p : freq) {
        freqToNums[p.second].insert(p.first);
    }

    vector<int> result;
    int count = 0;

    for (auto &p : freqToNums) {
        for (int num : p.second) {
            if (count < k) {
                result.push_back(num);
                count++;
            } else {
                break;
            }
        }
        if (count >= k) break;
    }

    return result;

    }
};
